Organ fixes and tweaks

DeeUnderscore opened the pull request:
## Limb removal and reattachment
- External organs now retain their `implants` and `internal_organs`. This makes it easier to add them back to the mob after the limb is reattached, which is what happens now. This resolves #14008.
- Cracking amputated limbs open now sets `open` instead of `stage`. This means that if you crack an amputated external organ open to go fishing in it, and then reattach it, you'll have to cauterize the incision. This isn't fine grained (it doesn't keep track of ribcage sawing), but it makes more sense than `stage`.
## Implants
- Implants now remain in amputated limbs. Previously, while small items were supposed to stay in amputated limbs, it was the the small items that _didn't_. 
- Implants now register being removed when the limb they're in is cut off. Previously, death alarms worked even if you were beating the victim to death with his own leg with the death alarm in it. 
- Caveat: Implants in limbs do not consider themselves re-implanted when the limb is reattached. Handling this would require more extensive modification to the implant system, as implants handle both internal setup and interactive setup in the same proc. I left it be for now. 
## Stumps
- Stumps no longer share the wound list with the amputated limb. I suppose this was either an accidental copy-by-reference where copy-by-value was more appropriate, or someone actually wanted the wounds to be fixable on either the stump or the limb. Problem is, stumps are amputated before reattachment, and your newly reattached limb ended up with the stump's `lost_limb` wound.

DeeUnderscore committed:
Make amputated organs retain more data

* Amputated organs keep track of implants and internal organs
* Amputated organs keep track of child organs
* Organs are not set to ORGAN_CUT_AWAY unless they're actually removed
  from the amputated body part
* Use open instead of stage for fishing inside amputated limbs

DeeUnderscore committed:
Limb amputations handle implant devices

Implants behave as if they were removed if the limb they're in is
removed.

DeeUnderscore committed:
Stop stumps from getting amputated limb's wounds

Previously stumps would get wound list from amputated limb by reference.
This created problems when limbs were reattached later, as they would
end up with lost_limb wounds.

After this change, limbs retain their own wounds, which have to be
mended post-reattachment.

DeeUnderscore committed:
Add changelog

comma commented on code/modules/organs/organ_external.dm:
```diff
@@ -219,13 +243,21 @@
+			organ.replaced(owner, src)
+				
+		for(var/obj/implant in implants)
+			implant.forceMove(owner)
```
will that do the trick? Doesn't need no fancy procs like implanted(owner)?

PsiOmegaDelta commented:
Seems reasonable at a glance, but is there anything notifying implants that they've been re-attached again (tho not necessarily to their original owner)?

DeeUnderscore commented:
Not at this point.

Like I said above, implants do both interactive configuration and their own setup inside `implanted()`. Ideally, I'd like to separate the interactive configuration and implant setup into two steps, and then only do the setup on limb replacement. This involves more extensive changes, though, so I decided to open the PR without them.

comma commented:
I think for now just setting implanted and imp_in would do.

DeeUnderscore committed:
Add implant handling to limb reattachment

/obj/item/weapon/implant objects in limbs now set to implanted on limb
reattach.

comma commented:
:+1:
